downloader/src/main/kotlin/Downloader.kt

```diff
@@ -321,6 +361,25 @@ class Downloader {
         return DownloadResult(startTime, outputDirectory, sourceArtifact = target.sourceArtifact)
     }
 
+    private fun downloadPomArtifact(target: Package, outputDirectory: File): DownloadResult {
+        val pomUrl = target.binaryArtifact.url.replaceAfterLast('.', "pom")
+
+        log.info {
+            "Trying to download POM artifact for '${target.id}' from '$pomUrl'..."
+        }
+
+        val startTime = Instant.now()
+
+        try {
+            val pomName = pomUrl.substringAfterLast('/')
+            File(outputDirectory, pomName).writeBytes(URL(pomUrl).readBytes())
+        } catch (e: FileNotFoundException) {
+            throw DownloadException("Failed to download the Maven POM for '${target.id}'.")
+        }
+
+        return DownloadResult(startTime, outputDirectory)
```

Comment:
The `sourceArtifact` of the `DownloadResult` should be set to the POM path. Otherwise the `Provenance` object created from this result does not reference what was scanned.

Reply:
Good catch, done.